packages/cli/src/decorators/Redactable.ts

```diff
@@ -0,0 +1,40 @@
+import type { UserLike } from '@/eventbus/audit.types';
+
+function toRedactable(userLike: UserLike) {
+	return {
+		userId: userLike.id,
+		_email: userLike.email,
+		_firstName: userLike.firstName,
+		_lastName: userLike.lastName,
+		globalRole: userLike.role,
+	};
+}
+
+type FieldName = 'user' | 'inviter' | 'invitee';
+
+/**
+ * Mark redactable properties in a `UserLike` field in an `AuditorService`
+ * method arg. These properties will be later redacted by the log streaming
+ * destination based on user prefs. Relevant only for `n8n.audit.*` logs.
+ */
+export const Redactable =
+	(fieldName: FieldName = 'user'): MethodDecorator =>
+	(_target, _propertyName, propertyDescriptor: PropertyDescriptor) => {
+		const originalMethod = propertyDescriptor.value as Function;
+
+		type MethodArgs = Array<{ [fieldName: string]: UserLike }>;
+
+		propertyDescriptor.value = function (...args: MethodArgs) {
+			const index = args.findIndex((arg) => arg[fieldName] !== undefined);
+
+			const userLike = args[index]?.[fieldName];
```

Comment:
Is it a misconfiguration if there isn't an argument with the specified name? Would be nice to throw in that case to catch the bug as early as possible

Reply:
Added error if decorator set but no property found